server/node/routes.js

```diff
@@ -25,70 +26,64 @@ router.get('/', (req, res) => {
 /**
  * Stripe integration to accept all types of payments with 3 POST endpoints.
  *
- * 1. POST endpoint to create orders with all user information.
- * 2. POST endpoint to complete a payment immediately when a card is used.
- * For payments using Elements, Payment Request, Apple Pay, Google Pay, Microsoft Pay.
+ * 1. POST endpoint to create a PaymentIntent.
+ * 2. For payments using Elements, Payment Request, Apple Pay, Google Pay, Microsoft Pay
+ * the PaymentIntent is confirmed automatically with Stripe.js on the client-side.
  * 3. POST endpoint to be set as a webhook endpoint on your Stripe account.
- * It creates a charge as soon as a non-card payment source becomes chargeable.
+ * It confirms the PaymentIntent as soon as a non-card payment source becomes chargeable.
  */
 
-// Create an order on the backend.
-router.post('/orders', async (req, res, next) => {
-  let {currency, items, email, shipping, createIntent} = req.body;
+// Calculate total payment amount based on items in basket.
+const calculatePaymentAmount = async items => {
+  const productList = await products.list();
+  const skus = productList.data.reduce((a, c) => [...a, ...c.skus.data], []);
+  const total = items.reduce((a, c) => {
+    const sku = skus.filter(sku => sku.id === c.parent)[0];
+    return a + sku.price * c.quantity;
+  }, 0);
+  return total;
+};
+
+// Create the PaymentIntent on the backend.
+router.post('/payment_intents', async (req, res, next) => {
+  let {currency, items} = req.body;
+  const amount = await calculatePaymentAmount(items);
+
   try {
-    let order = await orders.create(currency, items, email, shipping, createIntent);
-    return res.status(200).json({order});
+    const paymentIntent = await stripe.paymentIntents.create({
+      amount,
+      currency,
+      payment_method_types: [
+        // 'ach_credit_transfer', // throws: error: "Invalid currency: eur. The payment method `ach_credit_transfer` only supports the following currencies: usd."
+        'alipay',
+        'bancontact',
+        'card',
+        'eps',
+        'ideal',
```

Comment:
Just tried running this and got an invalid request error for the `payment_method_types` param. I believe ideal is currently not supported by PMs yet (I was using a fresh test account)

Reply:
This was a problem on our end which has been fixed. Should be fine now. Please `git pull` and retry :)